src/python/pants/option/options_test.py

```diff
@@ -76,15 +78,22 @@ def create_options(
     env: dict[str, str] | None = None,
     config: dict[str, dict[str, Any]] | None = None,
     extra_scope_infos: list[ScopeInfo] | None = None,
-) -> Options:
-    options = Options.create(
-        env=env or {},
-        config=Config.load([FileContent("pants.toml", toml.dumps(config or {}).encode())]),
-        known_scope_infos=[*(ScopeInfo(scope) for scope in scopes), *(extra_scope_infos or ())],
-        args=["./pants", *(args or ())],
-    )
-    register_fn(options)
-    return options
+) -> Generator[Options, None, None]:
+    config_content = toml.dumps(config or {}).encode()
```

Comment:
We need this to be a contextmanager so it can actually write the config to disk, for the Rust config reader to access.

Reply:
Why does the Rust config reader need it on disk? Can't work in memory at all?

Reply:
It could be made to, but there is no user-facing need for that, it's just for testing. And to use it here it would have to be some in-memory type we can share across Rust and Python, so we'd have to make a bunch of changes to adapt FileContent that way, and it didn't seem worth it just for tests.

Reply:
See below - looks like I'll have to do this anyway... :( 

For now I will revert this part of the change.